federationapi/routing/send.go

```diff
@@ -34,27 +34,55 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	// Event was passed to the roomserver
+	MetricsOutcomeOK = "ok"
+	// Event failed to be processed
+	MetricsOutcomeFail = "fail"
+	// Event failed auth checks
+	MetricsOutcomeRejected = "rejected"
+	// Terminated the transaction
+	MetricsOutcomeFatal = "fatal"
+	// The event has missing auth_events we need to fetch
+	MetricsWorkMissingAuthEvents = "missing_auth_events"
+	// No work had to be done as we had all prev/auth events
+	MetricsWorkDirect = "direct"
+	// The event has missing prev_events we need to call /g_m_e for
+	MetricsWorkMissingPrevEvents = "missing_prev_events"
+)
+
 var (
 	pduCountTotal = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: "dendrite",
 			Subsystem: "federationapi",
 			Name:      "recv_pdus",
+			Help:      "Number of incoming PDUs from remote servers with labels for success",
 		},
-		[]string{"status"},
+		[]string{"status"}, // 'success' or 'total'
 	)
 	eduCountTotal = prometheus.NewCounter(
 		prometheus.CounterOpts{
 			Namespace: "dendrite",
 			Subsystem: "federationapi",
 			Name:      "recv_edus",
+			Help:      "Number of incoming EDUs from remote servers",
+		},
+	)
+	processEventSummary = prometheus.NewSummaryVec(
```

Comment:
Would this be better as a Histogram rather than a Summary? I think we have similar statistics in the roomserver so you should be able to look at what they are.

Reply:
I'm copying what state res does currently, which uses Summary. Looking at the docs, Histogram is useful when you know the ranges (how long things will take to complete) but Summary is good when you don't know. I don't think we know the timings yet, so I'll keep it as Summary for now.